### DISCUSSION TITLE
Packaging of the tool

### DISCUSSION THREAD

**Discussion by MokhFn** [Blockchain-Technology-Lab/consensus-decentralization#125] [Ideas]
Hi, 

While working on #120 for the project, i opted for using [zet.releaser](https://zestreleaser.readthedocs.io/en/latest/) command `fullrelease`. When merging a PR, this command will : 

- Update the changelog automatically (assuming you filled it)
- Create a new Tag
- Create a python package and upload it to PyPI

I implemented this withing [tox](https://tox.wiki/en/4.11.1/) environments.

The question is, if this was okay with the team ?
If so, the core maintainers should create a new Repository (and maybe even Organization) on PyPI for this project, name it, then create a token to be able to auth, when uploading packages.
This token needs to be added as a secret within Github, and i'll need the name of the secret (for example, i called it PYPI_TOKEN), that way i can use it from Actions securely.

I did a small release test that you can find [here](https://pypi.org/project/btl-cd-release-test/).

Thanks in advance, and please keep in mind that i remain open to any suggestions.
PS : The release test is only to confirm the gh actions perform as expected, the software won't work.

@dimkarakostas @LadyChristina @mtefagh

**Comment by dimkarakostas** [comment]
The tool is still in heavy development, so I'd rather wait until it is in a more stable setting before packaging.

**Reply by MokhFn** [reply]
Okay, i'll keep #120 and the related PR open untill further instructions.
I will add a manual changelog in the meanwhile if you think that could be helpful.
Please let me know when packaging becomes "of use".
